libraries/Microsoft.Bot.Builder.AI.Orchestrator/OrchestratorRecognizer.cs

```diff
@@ -317,53 +313,60 @@ private static JToken EntityResultToInstanceJObject(string text, Result result)
 
         private void TryScoreEntities(string text, RecognizerResult recognizerResult)
         {
-            if (!this.ScoreEntities)
+            if (this._resolver == null)
             {
                 return;
             }
 
-            var results = _resolver.Score(text, LabelType.Entity);
-            recognizerResult.Properties.Add(EntitiesProperty, results);
-
-            if (results.Any())
+            try
             {
-                if (recognizerResult.Entities == null)
-                {
-                    recognizerResult.Entities = new JObject();
-                }
+                var results = _resolver.Score(text, LabelType.Entity);
+                recognizerResult.Properties.Add(EntitiesProperty, results);
 
-                var entitiesResult = recognizerResult.Entities;
-                foreach (var result in results)
+                if (results.Any())
                 {
-                    // add value
-                    JToken values;
-                    if (!entitiesResult.TryGetValue(result.Label.Name, StringComparison.OrdinalIgnoreCase, out values))
+                    if (recognizerResult.Entities == null)
                     {
-                        values = new JArray();
-                        entitiesResult[result.Label.Name] = values;
+                        recognizerResult.Entities = new JObject();
                     }
 
-                    ((JArray)values).Add(EntityResultToJObject(text, result));
-
-                    // get/create $instance
-                    JToken instanceRoot;
-                    if (!recognizerResult.Entities.TryGetValue("$instance", StringComparison.OrdinalIgnoreCase, out instanceRoot))
+                    var entitiesResult = recognizerResult.Entities;
+                    foreach (var result in results)
                     {
-                        instanceRoot = new JObject();
-                        recognizerResult.Entities["$instance"] = instanceRoot;
-                    }
+                        // add value
+                        JToken values;
+                        if (!entitiesResult.TryGetValue(result.Label.Name, StringComparison.OrdinalIgnoreCase, out values))
+                        {
+                            values = new JArray();
+                            entitiesResult[result.Label.Name] = values;
+                        }
 
-                    // add instanceData
-                    JToken instanceData;
-                    if (!((JObject)instanceRoot).TryGetValue(result.Label.Name, StringComparison.OrdinalIgnoreCase, out instanceData))
-                    {
-                        instanceData = new JArray();
-                        instanceRoot[result.Label.Name] = instanceData;
-                    }
+                        ((JArray)values).Add(EntityResultToJObject(text, result));
```

Comment:
I know this line was moved from above, but in any case it seems like a potentially dangerous one. If we just instantiated a JArray, we are safe (i.e. if the `if` statement in line 338 was executed). However, if the `if statement` did not get executed, we got a `JToken` from the json payload. Given that `JToken` is more general than `JArray`, this cast could lead to an invalid cast. Do we guarantee that 100% of the time this will be a JArray? If yes, then why isn't the `values` variable in line 337 declared as a JArray instead? to handle empty entries? If this is all on purpose, some comments explaining the decision making behind this could be useful for future maintainers

Reply:
Will make the declaration to JArray.